r2dbc-spi-test/src/main/java/io/r2dbc/spi/test/MockRowMetadata.java

```diff
@@ -50,6 +49,15 @@ public ColumnMetadata getColumnMetadata(Object identifier) {
         return this.columnMetadatas;
     }
 
+    @Override
+    public Collection<String> getColumnNames() {
+        Set<String> columnNames = new HashSet<>();
```

Comment:
Let's make this a `LinkedHashSet` to retain order.

After having this change, we are good to merge the PR.